<launch>
  <!-- GDB Debug Option -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="gdb -x $(dirname)/gdb_settings.gdb --ex run --args" />

  <!-- Verbose Mode Option -->
  <arg name="info" default="$(arg debug)" />
  <arg unless="$(arg info)" name="command_args" value="" />
  <arg if="$(arg info)" name="command_args" value="--debug" />

  <arg name="arm_id" default="panda" />

  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find pick_and_place)/config/rosconsole.conf"/>

  <include file="$(find panda_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- If needed, broadcast static tf for robot root -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_world_panda" args="0 0 0 0 0 0  world panda_link0" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_world_camera" args="-0.3 -.20 0.60 -1.5 0.07 -1.91 world camera_rgb_optical_frame" />

  <!-- Play the rosbag that contains the pointcloud data -->
  <!-- <node pkg="pick_and_place" type="publish_pc_bag" name="publish_pc_bag" /> -->

  <!-- Params for 3D sensors config -->
  <!-- <rosparam command="load" file="$(find pick_and_place)/config/sensors_3d.yaml" /> -->

  <!-- Params for the octomap monitor -->
  <!-- <param name="robot_description_planning/shape_transform_cache_lookup_wait_time" value="0.5" />
  <param name="octomap_frame" type="string" value="camera_rgb_optical_frame" />
  <param name="octomap_resolution" type="double" value="0.025" /> -->
  <!-- <param name="max_range" type="double" value="5.0" /> -->

  <arg name="planning_plugin" value="ompl_interface/OMPLPlanner" />
  <arg name="planning_adapters" default="default_planner_request_adapters/AddTimeParameterization"/>

  <!-- added complexity, comment out for now
    default_planner_request_adapters/ResolveConstraintFrames
    default_planner_request_adapters/FixWorkspaceBounds
    default_planner_request_adapters/FixStartStateBounds
    default_planner_request_adapters/FixStartStateCollision
    default_planner_request_adapters/FixStartStatePathConstraints -->

  <arg name="start_state_max_bounds_error" default="0.1" />
  <arg name="jiggle_fraction" default="0.05" />

  <node name="generate_contact_plan" pkg="pick_and_place" type="generate_contact_plan" respawn="false" launch-prefix="$(arg launch_prefix)" output="screen">
    <param name="planning_plugin" value="$(arg planning_plugin)" />
    <param name="request_adapters" value="$(arg planning_adapters)" />
    <param name="start_state_max_bounds_error" value="$(arg start_state_max_bounds_error)" />
    <param name="jiggle_fraction" value="$(arg jiggle_fraction)" />
    <rosparam command="load" file="$(find panda_moveit_config)/config/ompl_planning.yaml" subst_value="true"/>
    <rosparam command="load" file="$(find panda_moveit_config)/config/kinematics.yaml" subst_value="true"/>
  </node>

</launch>